gobblin-core/src/main/java/gobblin/serde/HiveSerDeWrapper.java

```diff
@@ -0,0 +1,164 @@
+
+  public static final String SERDE_SERIALIZER_TYPE = SERDE_SERIALIZER_PREFIX + "type";
+  public static final String SERDE_SERIALIZER_INPUT_FORMAT_TYPE = SERDE_SERIALIZER_PREFIX + "input.format.type";
+  public static final String SERDE_SERIALIZER_OUTPUT_FORMAT_TYPE = SERDE_SERIALIZER_PREFIX + "output.format.type";
+  public static final String SERDE_SERIALIZER_FILE_EXTENSION = SERDE_SERIALIZER_PREFIX + "extension";
+
+  public static final String SERDE_DESERIALIZER_TYPE = SERDE_DESERIALIZER_PREFIX + "type";
+  public static final String SERDE_DESERIALIZER_INPUT_FORMAT_TYPE = SERDE_DESERIALIZER_PREFIX + "input.format.type";
+  public static final String SERDE_DESERIALIZER_OUTPUT_FORMAT_TYPE = SERDE_DESERIALIZER_PREFIX + "output.format.type";
+  public static final String SERDE_DESERIALIZER_FILE_EXTENSION = SERDE_DESERIALIZER_PREFIX + "extension";
+
+  public enum BuiltInHiveSerDe {
+
+    AVRO(AvroSerDe.class.getName(), AvroContainerInputFormat.class.getName(), AvroContainerOutputFormat.class.getName(),
+        WriterOutputFormat.AVRO.getExtension()),
+    ORC(OrcSerde.class.getName(), OrcInputFormat.class.getName(), OrcOutputFormat.class.getName(),
+        WriterOutputFormat.ORC.getExtension()),
+    PARQUET(ParquetHiveSerDe.class.getName(), MapredParquetInputFormat.class.getName(),
+        MapredParquetOutputFormat.class.getName(), WriterOutputFormat.PARQUET.getExtension());
+
+    private final String serDeClassName;
+    private final String inputFormatClassName;
+    private final String outputFormatClassName;
+    private final String extension;
+
+    private BuiltInHiveSerDe(String serDeClassName, String inputFormatClassName, String outputFormatClassName,
+        String extension) {
+      this.serDeClassName = serDeClassName;
+      this.inputFormatClassName = inputFormatClassName;
+      this.outputFormatClassName = outputFormatClassName;
+      this.extension = extension;
+    }
+
+    @Override
+    public String toString() {
+      return this.serDeClassName;
+    }
+  }
+
+  private Optional<SerDe> serDe = Optional.absent();
+  private final String serDeClassName;
+  private final String inputFormatClassName;
+  private final String outputFormatClassName;
+  private final String extension;
+
+  private HiveSerDeWrapper(BuiltInHiveSerDe hiveSerDe) {
+    this(hiveSerDe.serDeClassName, hiveSerDe.inputFormatClassName, hiveSerDe.outputFormatClassName,
+        hiveSerDe.extension);
+  }
+
+  private HiveSerDeWrapper(String serDeClassName, String inputFormatClassName, String outputFormatClassName,
+      String extension) {
+    this.serDeClassName = serDeClassName;
+    this.inputFormatClassName = inputFormatClassName;
+    this.outputFormatClassName = outputFormatClassName;
+    this.extension = extension;
+  }
+
+  public SerDe getSerDe() throws IOException {
+    if (!this.serDe.isPresent()) {
+      try {
+        this.serDe = Optional.of((SerDe) Class.forName(this.serDeClassName).newInstance());
```

Comment:
You can do `Serde.class.cast(Class.forName(this.serDeClassName).newInstance())`.